paddle/fluid/imperative/reducer.h

```diff
@@ -194,6 +204,14 @@ class Reducer {
   std::vector<std::shared_ptr<imperative::VarBase>> rebuild_vars_;
   std::vector<int64_t> rebuild_var_indices_;
   const std::vector<size_t> group_size_limits_;
+
+  // Following variables are to help unused vars
+  std::unordered_map<GradOpNode*, size_t> node_deps_;
+  std::unordered_map<VariableWrapper*, size_t> var_index_map_;
+  std::vector<size_t> unused_vars_;
+  bool has_marked_unused_vars_{false};
+  bool find_unused_vars_{false};
+  bool need_finalize_backward_{false};
```

Comment:
`need_finalize_backward_` is little bit obscure, how about `all_group_ready`

Reply:
ok, i will fix it.